lib/CMakeLists.txt

```diff
@@ -19,6 +19,26 @@
 add_compile_options(-Wno-shadow)
 set(BUILD_SHARED_LIBS 1)
 
+if(BUILD_TESTING)
+  include(FetchContent)
+  fetchcontent_declare(
+    Catch2
+    GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+    GIT_TAG v3.9.1
+  )
+  fetchcontent_makeavailable(Catch2)
+
```

Comment:
This is one of the ways the Catch docs recommend retrieving the source. Are there any concerns about using FetchContent for ATS? I'm guessing it's fine, but we don't use it elsewhere yet and wanted to know whether @JosiahWI or @cmcfarlen (or others of course) had any concerns.

Reply:
Thanks for the feedback @JosiahWI. To avoid fetching during builds I created a clone in lib/Catch2 like we have for lib/swoc, etc.